mllib/src/test/scala/org/apache/spark/ml/util/DefaultReadWriteTest.scala

```diff
@@ -107,24 +133,34 @@ trait DefaultReadWriteTest extends TempDirectory { self: Suite =>
     }
     val model = estimator.fit(dataset)
 
-    // Test Estimator save/load
-    val estimator2 = testDefaultReadWrite(estimator)
-    testParams.foreach { case (p, v) =>
-      val param = estimator.getParam(p)
-      assert(estimator.get(param).get === estimator2.get(param).get)
+    val testFunctions = if (testParams.contains("initialModel")) {
+      Map(("initialModel", checkModelData.asInstanceOf[(Any, Any) => Unit]))
```

Comment:
I don't think it's a good idea. The code is technically incorrect. At the very least, we could pass in a map of functions for each param that has a special check. Like adding an extra method argument `checkParamsFunctions: Map[String, (Any, Any) => Unit] = Map.empty`. I don't think it's an elegant solution but it should at least apply generically. I think it might be reasonable to override the equals method, but we should check with @jkbradley.